### PR TITLE
feat: Adding ImageExtension.resize

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Adds a helper method to `ImageExtension` to make it easier to resize an image.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/extensions/image.dart [hunk 1]
```diff
@@ -90,4 +90,26 @@ extension ImageExtension on Image {
...
+  Future<Image> resize(Vector2 newSize) async {
+    final recorder = PictureRecorder();
+    Canvas(recorder).drawImageRect(
+      this,
+      Rect.fromLTWH(
+        0,
+        0,
+        width.toDouble(),
+        height.toDouble(),
+      ),
```
Comment (luanpotter) [inline packages/flame/lib/src/extensions/image.dart:104]: re-use the extension getBoundingRect ?

File: packages/flame/lib/src/extensions/image.dart [hunk 2]
```diff
@@ -90,4 +90,26 @@ extension ImageExtension on Image {
     }
     return fromPixels(newPixelData, width, height);
   }
+
+  /// Resizes this image to the given [newSize].
```
Comment (luanpotter) [inline packages/flame/lib/src/extensions/image.dart:94]: probably worth adding a note here that this is one of those kinds of _expensive_ operations. ie not to be done on gameloops

File: packages/flame/lib/src/extensions/image.dart [hunk 3]
```diff
@@ -90,4 +90,26 @@ extension ImageExtension on Image {
...
+    final recorder = PictureRecorder();
+    Canvas(recorder).drawImageRect(
+      this,
+      Rect.fromLTWH(
+        0,
+        0,
+        width.toDouble(),
+        height.toDouble(),
+      ),
+      Rect.fromLTWH(0, 0, newSize.x, newSize.y),
```
Comment (luanpotter) [inline packages/flame/lib/src/extensions/image.dart:105]: similarly, newSize.toRect()

**Review by renancaraujo — APPROVED** [review]

**Review by spydon — APPROVED** [review]
Lgtm, one minor comment

File: examples/lib/stories/image/image.dart [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            context.numberProperty('width', 200),
+            context.numberProperty('height', 300),
+          ),
+        ),
+      ),
+      codeLink: baseLink('image/resize.dart'),
+      info: '''
+        Shows how a dart:ui `Image` can be resized using Flame Image extensions.
+        Use the properties on the side to change the size of the image.
+      ''',
```
[the remark this answers is not in the document]
Comment (erickzanardo) [reply examples/lib/stories/image/image.dart:24]: Opsie, yes, it actually should

File: packages/flame/lib/src/extensions/image.dart [hunk 5]
```diff
@@ -90,4 +90,25 @@ extension ImageExtension on Image {
     }
     return fromPixels(newPixelData, width, height);
   }
+
+  /// Resizes this image to the given [newSize].
+  ///
+  /// Keep in mind that is considered an expensive operation and should be
+  /// avoided being called in the game loop methods, preferable using it
```
Comment (luanpotter) [inline packages/flame/lib/src/extensions/image.dart:97]:
```suggestion
  /// avoided in the the game loop methods. Prefer using it
```

**Review by luanpotter — APPROVED** [review]
LGTM

**Pull request merged**